src/modules/0.0.21/aws_codebuild/mappers/build.ts

```diff
@@ -0,0 +1,220 @@
+        const outB = await this.buildListMapper(build, ctx);
+        if (outB) out.push(outB);
+      }
+      return out;
+    },
+    updateOrReplace: () => 'update',
+    update: async (es: CodebuildBuildList[], ctx: Context) => {
+      // Right now we can only modify AWS-generated fields in the database.
+      // This implies that on `update`s we only have to restore the values for those records.
+      const out = [];
+      for (const e of es) {
+        const cloudRecord = ctx?.memo?.cloud?.CodebuildBuildList?.[e.awsId];
+        await this.module.buildList.db.update(cloudRecord, ctx);
+        out.push(cloudRecord);
+      }
+      return out;
+    },
+    delete: async (bds: CodebuildBuildList[], ctx: Context) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+      const idsToStop = bds.filter(bd => bd.buildStatus === BuildStatus.IN_PROGRESS).map(bd => bd.awsId);
+      for (const id of idsToStop) {
+        const input: StopBuildInput = { id };
+        await this.stopBuild(client.cbClient, input);
+      }
+      if (idsToStop.length > 0) await this.waitForBuildsToStop(client.cbClient, idsToStop);
+      let idsToDel = bds.map(bd => bd.awsId);
+      // Wait for ~2.5min until builds can be deleted
+      let i = 0;
+      do {
+        const input: BatchDeleteBuildsCommandInput = { ids: idsToDel };
+        const out = await this.deleteBuilds(client.cbClient, input);
+        if (out?.buildsNotDeleted?.length === 0) break;
+        idsToDel = out?.buildsNotDeleted?.map(bd => bd.id as string) ?? [];
+        await new Promise(r => setTimeout(r, 5000)); // Sleep for 5s
+        i++;
+      } while (i < 60 || idsToDel.length > 0);
+      if (i === 59) throw new Error('Error deleting builds');
+    },
+  });
+
+  constructor(module: AwsCodebuildModule) {
+    super();
+    this.module = module;
+    super.init();
+  }
+}
+
+export class CodebuildBuildImportMapper extends MapperBase<CodebuildBuildImport> {
+  module: AwsCodebuildModule;
+  entity = CodebuildBuildImport;
+  entityId = (e: CodebuildBuildImport) => e.id.toString();
+  equals = (a: CodebuildBuildImport, b: CodebuildBuildImport) =>
+    Object.is(a.id, b.id) && Object.is(a.project.projectName, b.project.projectName);
+
+  startBuild = crudBuilderFormat<CodeBuild, 'startBuild', Build | undefined>(
+    'startBuild',
+    input => input,
+    res => res?.build,
+  );
+
+  db = new Crud2<CodebuildBuildImport>({
+    create: (es: CodebuildBuildImport[], ctx: Context) => ctx.orm.save(CodebuildBuildImport, es),
+    update: (es: CodebuildBuildImport[], ctx: Context) => ctx.orm.save(CodebuildBuildImport, es),
+    delete: (es: CodebuildBuildImport[], ctx: Context) => ctx.orm.remove(CodebuildBuildImport, es),
+    read: async (ctx: Context, id?: string) => {
+      const opts = id
+        ? {
+            where: {
+              id,
+            },
+          }
+        : {};
+      return await ctx.orm.find(CodebuildBuildImport, opts);
+    },
+  });
```

Comment:
isn't this the default implementation for the `db.read`?

Reply:
hmm I think so, I copied it over from ACM modules since that was the only reference point I had for it. I can try removing it here, but we should probably remove it there?

Reply:
I've just seen the default implementation, it is 
```
where: {
                  [cloudColumn]: id,
                },
```
so it would depend if in his case we are referring to the db `id` or `id` is the `cloudColumn`?

Reply:
I was able to remove it!